src/main/java/frc/robot/subsystems/indexer/IndexerSubsystem.java

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems.indexer;
+
+import java.util.function.DoubleSupplier;
+
+import org.littletonrobotics.junction.Logger;
+
+import com.revrobotics.RelativeEncoder;
+import com.revrobotics.spark.SparkClosedLoopController;
+import com.revrobotics.spark.SparkMax;
+import com.revrobotics.spark.SparkSim;
+import com.revrobotics.spark.SparkBase.ControlType;
+import com.revrobotics.spark.SparkBase.PersistMode;
+import com.revrobotics.spark.SparkBase.ResetMode;
+
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.wpilibj.DigitalOutput;
+import edu.wpi.first.wpilibj.RobotController;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.Commands;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Configs;
+import frc.robot.Constants.IndexerConstants;
+
+public class IndexerSubsystem extends SubsystemBase {
+  private final SparkMax indexerSpark;
+  private final SparkClosedLoopController controller;
+  private final RelativeEncoder encoder;
+
+  public final DigitalOutput intakeSensorTrigger, launcherSensorTrigger;
+  public final IndexerSensor intakeSensor, launcherSensor;
+
+  private final SparkSim indexerSparkSim; // In case we ever need to simulate the motor
+
+  /**
+   * Creates a new IndexerSubsystem
+   */
+  public IndexerSubsystem() {
+    indexerSpark = new SparkMax(IndexerConstants.kMotorPort, IndexerConstants.kMotorType);
+
+    indexerSparkSim = new SparkSim(indexerSpark, DCMotor.getNeo550(1));
```

Comment:
If we’re going to use sim. Please use interfaces so we can pass in an instance of SparkMax OR SparkSim into the constructor of this class.